perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_attribute_def_def_unixGID_namespace.java

```diff
@@ -133,34 +143,25 @@ public void checkAttributeSemantics(PerunSessionImpl sess, Group group, Attribut
 				for(String namespace : namespacesWhereGroupMustHaveGIDifItHaveUnixNameThere) {
 					Attribute unixGroupName = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, group, A_G_unixGroupName_namespace + ":" + namespace);
 					if(unixGroupName.getValue() != null) {
-						throw new WrongAttributeValueException(attribute, group, "Group is propagated to the facility where it have set unix group name so it must have unix GID too.");
+						throw new WrongReferenceAttributeValueException(attribute,  unixGroupName, group, null, group, null, "Group is propagated to the facility where it have set unix group name so it must have unix GID too.");
 					}
 				}
 				return;   //Group is not propagated to any facility in this GID namespace or it doesn't have set unix name there so it doesn't need to have unix GID.
 			}
 
-			//Special behaviour if gid is null
-			Integer attrValue;
-			if(attribute.getValue() == null) {
-				throw new WrongAttributeValueException(attribute, group, "Unix GID must be set");
-			} else {
-				attrValue = (Integer) attribute.getValue();
-			}
+			Integer attrValue = attribute.valueAsInteger();
```

Comment:
This will not throw an exception if value will be null there so it will continue even if it shouldn't.

Reply:
But I thought that if the value was null, the method would end due to previous condition, would it not?

Reply:
Yeah, you are right. If value will be null, it will end with an exception or by return. So this is ok.